MapleServer2/PacketHandlers/Game/UserChatHandler.cs

```diff
@@ -21,7 +22,44 @@ public override void Handle(GameSession session, PacketReader packet)
             packet.ReadLong();
 
             GameCommandActions.Process(session, message);
-            session.FieldManager.SendChat(session.Player, message, type);
+
+            switch(type)
+            {
+                case ChatType.Channel:
+                    //TODO: Send to all players on current channel
+                case ChatType.World:
+                    //Send to all players online
+                    MapleServer.BroadcastPacketAll(ChatPacket.Send(session.Player, message, type));
+                    break;
+                case ChatType.GuildNotice:
+                case ChatType.Guild:
+                    //TODO: Send to all in guild
+                    break;
+                case ChatType.Party:
+                    //TODO: Send to all in party
+                    break;
+                case ChatType.WhisperTo:
+                    Types.Player recipientPlayer = null;
+                    MapleServer.BroadcastAll(pSession => {
+                        if (pSession.Player.Name == recipient)
+                        {
+                            pSession.Send(ChatPacket.Send(session.Player, message, ChatType.WhisperFrom));
```

Comment:
The player who sent the whisper should only get the packet if the recipient was found, so maybe move this down to line 52 and send both packets together?